src/factory.js

```diff
@@ -61,48 +61,10 @@ export default function plotComponentFactory(Plotly) {
       this.getRef = this.getRef.bind(this);
       this.handleUpdate = this.handleUpdate.bind(this);
       this.figureCallback = this.figureCallback.bind(this);
+      this.updatePlotly = this.updatePlotly.bind(this);
     }
 
-    componentDidMount() {
-      this.unmounting = false;
-
-      this.p = this.p
-        .then(() => {
-          if (!this.el) {
-            let error;
-            if (this.unmounting) {
-              error = new Error('Component is unmounting');
-              error.reason = 'unmounting';
-            } else {
-              error = new Error('Missing element reference');
-            }
-            throw error;
-          }
-          return Plotly.newPlot(this.el, {
-            data: this.props.data,
-            layout: this.props.layout,
-            config: this.props.config,
-            frames: this.props.frames,
-          });
-        })
-        .then(() => this.syncWindowResize(null, true))
-        .then(this.syncEventHandlers)
-        .then(this.attachUpdateEvents)
-        .then(() => this.figureCallback(this.props.onInitialized))
-        .catch(err => {
-          if (err.reason === 'unmounting') {
-            return;
-          }
-          console.error('Error while plotting:', err); // eslint-disable-line no-console
-          if (this.props.onError) {
-            this.props.onError(err);
-          }
-        });
-    }
-
-    UNSAFE_componentWillUpdate(nextProps) {
-      this.unmounting = false;
-
+    shouldComponentUpdate(nextProps) {
```

Comment:
I have hesitations about this... It's been suggested a bunch of times that we use this BUT the docs basically say not to rely on it to prevent updates. So I don't think we should: https://reactjs.org/docs/react-component.html#shouldcomponentupdate

Reply:
Yep

> In the future React may treat shouldComponentUpdate() as a hint rather than a strict directive